src/Mod/Part/App/TopoShape.h

```diff
@@ -1223,25 +1385,120 @@ class PartExport TopoShape: public Data::ComplexGeoData
+    {
+#if OCC_VERSION_HEX >= 0x070800
+        return std::hash<TopoDS_Shape> {}(s);
+#else
+        return s.HashCode(INT_MAX);
+#endif
+    }
+    inline bool operator()(const TopoShape& a, const TopoShape& b) const
+    {
+        return a.getShape().IsSame(b.getShape());
+    }
+    inline bool operator()(const TopoDS_Shape& a, const TopoDS_Shape& b) const
+    {
+        return a.IsSame(b);
+    }
+    template<class T>
+    static inline void hash_combine(std::size_t& seed, const T& v)
+    {
+        // copied from boost::hash_combine
+        std::hash<T> hasher;
+        seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+    }
+    inline size_t operator()(const std::pair<TopoShape, TopoShape>& s) const
+    {
+#if OCC_VERSION_HEX >= 0x070800
+        size_t res = std::hash<TopoDS_Shape> {}(s.first.getShape());
+        hash_combine(res, std::hash<TopoDS_Shape> {}(s.second.getShape()));
+#else
+        size_t res = s.first.getShape().HashCode(INT_MAX);
+        hash_combine(res, s.second.getShape().HashCode(INT_MAX));
+#endif
+        return res;
+    }
+    inline size_t operator()(const std::pair<TopoDS_Shape, TopoDS_Shape>& s) const
+    {
+#if OCC_VERSION_HEX >= 0x070800
+        size_t res = std::hash<TopoDS_Shape> {}(s.first);
+        hash_combine(res, std::hash<TopoDS_Shape> {}(s.second));
+#else
+        size_t res = s.first.HashCode(INT_MAX);
+        hash_combine(res, s.second.HashCode(INT_MAX));
+#endif
+        return res;
+    }
+    inline bool operator()(const std::pair<TopoShape, TopoShape>& a,
+                           const std::pair<TopoShape, TopoShape>& b) const
+    {
+        return a.first.getShape().IsSame(b.first.getShape())
+            && a.second.getShape().IsSame(b.second.getShape());
+    }
+    inline bool operator()(const std::pair<TopoDS_Shape, TopoDS_Shape>& a,
+                           const std::pair<TopoDS_Shape, TopoDS_Shape>& b) const
+    {
+        return a.first.IsSame(b.first) && a.second.IsSame(b.second);
+    }
+};
+
 /** Shape mapper for generic BRepBuilderAPI_MakeShape derived class
  *
  * Uses BRepBuilderAPI_MakeShape::Modified/Generated() function to extract
  * shape history for generating mapped element names
  */
-struct PartExport MapperMaker: TopoShape::Mapper
-{
-    BRepBuilderAPI_MakeShape& maker;
-    explicit MapperMaker(BRepBuilderAPI_MakeShape& maker)
-        : maker(maker)
+struct PartExport MapperMaker: TopoShape::Mapper {
+    BRepBuilderAPI_MakeShape &maker;
+    MapperMaker(BRepBuilderAPI_MakeShape &maker)
+        :maker(maker)
     {}
-    const std::vector<TopoDS_Shape>& modified(const TopoDS_Shape& s) const override;
-    const std::vector<TopoDS_Shape>& generated(const TopoDS_Shape& s) const override;
+    virtual const std::vector<TopoDS_Shape> &modified(const TopoDS_Shape &s) const override;
+    virtual const std::vector<TopoDS_Shape> &generated(const TopoDS_Shape &s) const override;
 };

```

Comment:
These three lint fixes, maybe?  And the ones below?  I've been following the lint recommendations on adding explicit and removing virtual.

Reply:
Missed these lint warning because on the previous comments I ran clang-tidy only on `TopoShapeExpansion.cpp` test file

Should be ok in the next commit